edweiss-app/components/todo/abstractTodoEditor.tsx

```diff
@@ -56,25 +56,25 @@ export const AbstractTodoEditor: React.FC<{
     const [showPickerDate, setShowPickerDate] = useState(false);
     const [showPickerTime, setShowPickerTime] = useState(false);
 
-    // Constants    
-    const downButtonTitle = (editable) ? t(`todo:edit_button`) : t(`todo:save_button`);
-    const downButtonIconName = (editable) ? "create" : "save";
-    const screenTitle = (editable) ? t(`todo:edit_header`) : t(`todo:create_header`);
-    const downButtonAction = (editable) ? editTodoAction : saveAction;
-
-
     // Toogle the save button only when valid
-    const modifiedTodo = !(editable && todo) ? undefined :
-        {
-            name: name, description: description === "" ? undefined : description, status: status,
-            dueDate: (!providedDate && !(dateChanged || timeChanged)) ? undefined : date
-        };
-    // const isInvalid = (editable) ? sameTodos(todo!, modifiedTodo!) : name === ""; // FIX ME !
     const isInvalid = name === "";
 
+    // Constants for save state
+    let downButtonIconName: "create" | "save" = "save";
+    let downButtonTitle = t(`todo:save_button`);
+    let screenTitle = t(`todo:create_header`);
+    let downButtonAction = saveAction;
+
+    if (editable) {
```

Comment:
You can change the initialisation of this variables to doesn't need to change them later, by exemple: 
`const downButtonIconName = editable ? "create" : "save";
const screenTitle = editable ? t(`todo:edit_header`) : t(`todo:create_header`);
const downButtonAction = editable ? editTodoAction : saveAction;
`

Reply:
That's super correct, and to be completly honest with you I have a preference for this approach using ternary operators. In the previous codebase version I even being using it. However, Sonar Clound complained about it and needed to reduce the code complexity. (Each ternary operator introduce one point of complexity). So it 'a contre-coeur' that I had to do this if else statement. Hope you understand.